cpu/cc2538/radio/cc2538_rf_netdev.c

```diff
@@ -0,0 +1,382 @@
+    mutex_lock(&dev->mutex);
+
+    /* Flush TX FIFO */
+    RFCORE_SFR_RFST = ISFLUSHTX;
+
+    /* The first byte of the TX FIFO must be the packet length,
+       but we don't know what it is yet. Write a null byte to the
+       start of the FIFO, so we can come back and update it later */
+    rfcore_write_byte(0);
+
+    for (int i = 0; i < count; i++) {
+        pkt_len += vector[i].iov_len;
+
+        if (pkt_len > CC2538_RF_MAX_DATA_LEN) {
+            DEBUG("cc2538_rf: packet too large (%u > %u)\n",
+                  pkt_len, CC2538_RF_MAX_DATA_LEN);
+            return -EOVERFLOW;
+        }
+
+        rfcore_write_fifo(vector[i].iov_base, vector[i].iov_len);
+    }
+
+    /* Set first byte of TX FIFO to the packet length */
+    rfcore_poke_tx_fifo(0, pkt_len + CC2538_AUTOCRC_LEN);
+
+    RFCORE_SFR_RFST = ISTXON;
+    mutex_unlock(&dev->mutex);
+
+    return pkt_len;
+}
+
+static int _recv(netdev2_t *netdev, char *buf, int len, void *info)
+{
+    cc2538_rf_t *dev = (cc2538_rf_t *) netdev;
+    size_t pkt_len;
+
+    mutex_lock(&dev->mutex);
+
+    if (buf == NULL) {
+        /* GNRC wants to know how much data we've got for it */
+        pkt_len = rfcore_read_byte();
+
+        /* If the value of the first byte of the RX FIFO does not correspond
+           to the amount of data received, then drop the packet. */
+        if (pkt_len != RFCORE_XREG_RXFIFOCNT) {
+            RFCORE_SFR_RFST = ISFLUSHRX;
+            mutex_unlock(&dev->mutex);
+            return -EOVERFLOW;
+        }
+
+        if (len > 0) {
+            /* GNRC wants us to drop the packet */
+            RFCORE_SFR_RFST = ISFLUSHRX;
+        }
+
+        mutex_unlock(&dev->mutex);
+        return pkt_len - IEEE802154_FCS_LEN;
+    }
+    else {
+        /* GNRC is expecting len bytes of data */
+        pkt_len = len;
+    }
+
+    rfcore_read_fifo(buf, pkt_len);
+
+    if (info != NULL) {
+        netdev2_ieee802154_rx_info_t *radio_info = info;
+        radio_info->rssi = rfcore_read_byte();
+
+        /* This is not strictly 802.15.4 compliant, since according to
+           the CC2538 documentation, this value will tend between ~50
+           for the lowest quality link detectable by the receiver, and
+           ~110 for the highest. The IEEE 802.15.4 spec mandates that
+           this value be between 0-255, with 0 as lowest quality and
+           255 as the highest. FIXME. */
+        radio_info->lqi = rfcore_read_byte() & 0x7F;
+    }
+
+    RFCORE_SFR_RFST = ISFLUSHRX;
+    mutex_unlock(&dev->mutex);
+
+    return pkt_len;
+}
+
+static void _isr(netdev2_t *netdev)
+{
+    netdev->event_callback(netdev, NETDEV2_EVENT_RX_COMPLETE);
+}
+
+static int _init(netdev2_t *netdev)
+{
+    cc2538_rf_t *dev = (cc2538_rf_t *) netdev;
+    _dev = netdev;
+
+    mutex_lock(&dev->mutex);
+
+    uint16_t pan = cc2538_get_pan();
+    uint16_t chan = cc2538_get_chan();
+    uint16_t addr_short = cc2538_get_addr_short();
+    uint64_t addr_long = cc2538_get_addr_long();
```

Comment:
These variables are only used for updating the `netdev2` struct with whatever the hardware was already configured as during driver init - in this case the long address will already be set to a unique EUI-64 address.